pkg/reconciler/wrappedPod.go

```diff
@@ -71,10 +71,9 @@ func getFlatIPSet(pod v1.Pod) map[string]void {
 			continue
 		}
 
-		if network.Interface[:multusPrefixSize] == multusInterfaceNamePrefix {
-			for _, ip := range network.IPs {
-				ipSet[ip] = empty
-			}
+		for _, ip := range network.IPs {
```

Comment:
What if the secondary interface is not managed by IPAM CNI whereabouts? Such as dhcp, host-local...

Reply:
This is a valid concern, but, I think it is not introduced by this PR. 

Furthermore, iiuc, the scenario you point out @carezkh would only trigger something nasty (the reconciler would think IP A is still in use, when in truth it is was served by another IPAM plugin) **if** the user is using the same subnet in multiple networks, something which I think cannot be a use case. Am I right ? If so, we could probably get away with documenting this "limitation".

Reply:
In my opinion, user can use the same subnet in multiple networks when **the underlay networks of multiple NetworkAttachmentDefinitions are isolated** (maybe different storage networks or VLANs).

Example like below,
`cat macvlan-whereabouts1.yaml`

> apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: macvlan-whereabouts1
spec:
  config: '{
      "cniVersion": "0.3.1",
      "type": "macvlan",
      "master": "eth0",
      "mode": "bridge",
      "ipam": {
          "type": "whereabouts",
          "range": "192.168.0.100/24",
          "gateway": "192.168.0.1"
      }
    }'

`cat macvlan-whereabouts2.yaml`

> apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: macvlan-whereabouts2
spec:
  config: '{
      "cniVersion": "0.3.1",
      "type": "macvlan",
      "master": "eth1",
      "mode": "bridge",
      "ipam": {
          "type": "whereabouts",
          "range": "192.168.0.100/24",
          "gateway": "192.168.0.1"
      }
    }'

If the underlay networks of interface `eth0` and `eth1` are different storage network, the same subnet is ok. Am i rignt?

However, it's a little hard to tell if the network is isolated, maybe globally unique IP address is ok.

Reply:
Given that we (at least I) have no clue as to how to detect (via any sort of api) if the underlying interfaces are connected, and the mess the routing tables would like in the pods, I personally think the safest course of action is to assume globally unique IP addresses and document that.

I don't think this breaks the CNI spec, but I'm unsure about the de-facto standard for Kubernetes Network Custom Resource Definition.

Reply:
If the mess of the routing tables (in pods) caused by imporper use is inevitable, i think it should be handled by net adminitrator instead of CNI plugin.

Net admin plans and deploys the network and instructs others in its proper use, and IPAM CNI can assign IP addresses from pool without caring about globally IP overlapping. 

Anyway, the globally unique IP addresses (what this PR has done) is ok.

Reply:
> What if the secondary interface is not managed by IPAM CNI whereabouts? Such as dhcp, host-local...

I think that's true, but that issue would still exist without this fix. The "net" naming is coming from multus and not IPAM. So without this fix, the ip-reconciler is broken and a blocker as it automatically cleans up any secondary Multus interface not named "net". Which results in whereabouts assigning duplicate IPs and IP pool never filling up.

I think the point you raised is a separate issue with ip-reconciler

Reply:
I agree.